cart/domain/cart/discount.go

```diff
@@ -128,12 +164,7 @@ func mapToSlice(collection map[string]*AppliedDiscount) []AppliedDiscount {
 }
 
 // mapDiscounts map type + title of discount to corresponding discount
-func mapDiscounts(result map[string]*AppliedDiscount, discountable WithDiscount) (map[string]*AppliedDiscount, error) {
-	discounts, err := discountable.MergeDiscounts()
-	// in case discounts cannot be collected, stop execution
-	if err != nil {
-		return nil, err
-	}
+func mapDiscounts(result map[string]*AppliedDiscount, discounts AppliedDiscounts) (map[string]*AppliedDiscount, error) {
 	for _, discount := range discounts {
```

Comment:
I think this code can still be left in the function by passing an interface. Otherwise this code is the same for every place it is called